packages/ws/src/ws/WebSocketShard.ts

```diff
@@ -393,7 +452,9 @@ export class WebSocketShard extends AsyncEventEmitter<WebSocketShardEventsMap> {
 			this.inflate.push(Buffer.from(decompressable), flush ? zlib.Z_SYNC_FLUSH : zlib.Z_NO_FLUSH);
 
 			if (this.inflate.err) {
-				this.emit('error', `${this.inflate.err}${this.inflate.msg ? `: ${this.inflate.msg}` : ''}`);
+				this.emit(WebSocketShardEvents.Error, {
+					error: `${this.inflate.err}${this.inflate.msg ? `: ${this.inflate.msg}` : ''}`,
```

Comment:
Maybe this it a bit outside of scope, maybe it's not, but anyway:
Shouldn't this be an `Error`? Throwing a string (see line 288) does not seem right to me.

Reply:
You might be right, could construct an `Error` object here since its the only place that makes us use `unknown`/`any` for the event.